romanesco/__init__.py

```diff
@@ -199,258 +152,97 @@ def run(analysis, inputs, outputs=None, auto_convert=True, validate=True):
         no validation.
     :returns: A dictionary of the form ``name: binding`` where ``name`` is
         the name of the output and ``binding`` is an output binding of the form
-        ``{"format": format, "data": data}``. If the `outputs` parameter
+        ``{"format": format, "data": data}``. If the `outputs` param
         is specified, the formats of these bindings will match those given in
         `outputs`. Additionally, ``"data"`` may be absent if an output URI
         was provided. Instead, those outputs will be saved to that URI and
         the output binding will contain the location in the ``"uri"`` field.
     """
+    def extractId(spec):
+        return spec["id"] if "id" in spec else spec["name"]
```

Comment:
Ah, I see this is how you are maintaining backwards compatibility. Was wondering how it worked against unmodified tangelohub. Nice.